src/SDKs/Batch/DataPlane/Azure.Batch.Unit.Tests/BatchSharedKeyCredentialUnitTests.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace Azure.Batch.Unit.Tests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Net.Http;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Batch.Protocol;
+    using Microsoft.Net.Http.Server;
+    using Xunit;
+
+    public class BatchSharedKeyCredentialUnitTests
+    {
+        private const string url = "http://localhost:2055";
+
+        [Theory]
+        [MemberData(nameof(HttpMethods))]
+        public async Task TestSharedKeySigning_ContentLength(HttpMethod httpMethod)
+        {
+            BatchSharedKeyCredential creds = new BatchSharedKeyCredential(ClientUnitTestCommon.DummyAccountName, ClientUnitTestCommon.DummyAccountKey);
+
+            HttpRequestMessage message = new HttpRequestMessage(httpMethod, url);
+            message.Headers.Add("client-request-id", Guid.NewGuid().ToString());
+
+            await creds.ProcessHttpRequestAsync(message, CancellationToken.None);
+            Assert.NotNull(message.Headers.Authorization);
+
+            void AssertLambda(RequestContext ctx)
+            {
+                if (httpMethod == HttpMethod.Head || httpMethod == HttpMethod.Get)
```

Comment:
Is HttpMethod an enum?  If so consider using a switch instead of a series of if-clauses.